sys/include/net/gnrc/tcp.h

```diff
@@ -34,6 +34,39 @@
 extern "C" {
 #endif
 
+
+/**
+ * @brief Address information for a single TCP connection endpoint.
+ * @note Must be compatible to sock_tcp_ep_t.
+ */
+typedef struct {
+    int family; /**< IP Address family. GNRC supports currently only IPv6. */
+
+    union {
+#ifdef MODULE_GNRC_IPV6
+        uint8_t ipv6[sizeof(ipv6_addr_t)]; /**< IPv6 Address storage */
+#endif
+    } addr;
+
+    uint16_t netif; /**< Network interface ID */
+    uint16_t port;  /**< Port number (in host byte order) */
+} gnrc_tcp_ep_t;
+
+
+/**
+ * @brief Initialize TCP connection endpoint.
+ *
+ * @param[in,out] ep               Endpoint to initialize.
+ * @param[in]     address_family   Address family of @p address.
+ * @param[in]     address          Address information for endpoint.
+ * @param[in]     port             Port number for endpoint.
+ *
+ * @returns   0 on success.
+ *            -EAFNOSUPPORT if @p address_family is not supported.
+ *            -EINVAL if parsing of @p address failed.
+ */
+int gnrc_tcp_ep_init(gnrc_tcp_ep_t *ep, int address_family, char *address, uint16_t port);
```

Comment:
Given, that the address is apparently a string, I find the naming somewhat unfortunate, as it doesn't really initialize the struct, but actually does some conversion to the given parameters. I would prefer to do this outside or have something similar to `sock_util`'s [`sock_udp_str2ep()`](https://doc.riot-os.org/group__net__sock__util.html#gac1f524a66e0ff64589ef36dd53b817ee) so only have a string as input parameter.

Reply:
Ping?